ttnn/cpp/ttnn/operations/pool/maxpool/maxpool_pybind.hpp

```diff
@@ -0,0 +1,132 @@
+        | dilation_w        | kernel dilation in width dim  | uint32_t      |             | No       |
+        | memory_config     | Output memory config          | MemoryConfig  |             | No       |
+        +-------------------+-------------------------------+---------------+-------------+----------+
+    )doc");
+
+    module.def(
+        "max_pool2d_v2",
+        &max_pool2d_v2,
+        py::arg("input").noconvert(),
+        py::arg("reader_indices").noconvert(),
+        py::arg("in_n").noconvert(),
+        py::arg("in_h").noconvert(),
+        py::arg("in_w").noconvert(),
+        py::arg("kernel_h").noconvert(),
+        py::arg("kernel_w").noconvert(),
+        py::arg("stride_h") = 1,
+        py::arg("stride_w") = 1,
+        py::arg("pad_h") = 0,
+        py::arg("pad_w") = 0,
+        py::arg("dilation_h") = 1,
+        py::arg("dilation_w") = 1,
+        py::kw_only(),
+        py::arg("memory_config") = operation::DEFAULT_OUTPUT_MEMORY_CONFIG,
+        py::arg("nblocks") = 1,
+        py::arg("use_multicore") = true,
+        R"doc(
+        Max Pool 2D
+        +-------------------+-------------------------------+---------------+-------------+----------+
+        | Argument          | Description                   | Data type     | Valid range | Required |
+        +===================+===============================+===============+=============+==========+
+        | input             | Input activations tensor      | Tensor        |             | Yes      |
+        | in_n              | Input nbatch                  | Tensor        |             | Yes      |
+        | in_h              | Input height                  | Tensor        |             | Yes      |
+        | in_w              | Input width                   | Tensor        |             | Yes      |
+        | kernel_h          | kernel window height          | uint32_t      |             | Yes      |
+        | kernel_w          | kernel window width           | uint32_t      |             | Yes      |
+        | stride_h          | stride in height dim          | uint32_t      |             | No       |
+        | stride_w          | stride in width dim           | uint32_t      |             | No       |
+        | pad_h             | padding in height dim         | uint32_t      |             | No       |
+        | pad_w             | padding in width dim          | uint32_t      |             | No       |
+        | dilation_h        | kernel dilation in height dim | uint32_t      |             | No       |
+        | dilation_w        | kernel dilation in width dim  | uint32_t      |             | No       |
+        | memory_config     | output tensor memory config   | MemoryConfig  |             | No       |
+        +-------------------+-------------------------------+---------------+-------------+----------+
+    )doc");
+    module.def(
+        "maxpool2d",
```

Comment:
why inconsistent name with `max_pool_2d`?

Reply:
Lets be serious here, it is not a zoo

- max_pool2d
- max_pool2d_v2
- maxpool2d

Reply:
Maxpool has multiple versions currently, and is a WIP. This will be cleaned up later.

Reply:
Is it possible to use the same equivalent PyTorch name and simply overload this?   Are the arguments truly identical here?

Reply:
Which versions are in use? I am guessing the old tests use the older ones. Are they still needed?

Also, having different implementations based on `max_pool2d` and `maxpool2d` is unacceptable IMO. I think at the very least it should be `max_pool2d_deprecated` or `max_pool2d_legacy` or something...

Reply:
I've cleaned it up currently to 

- `max_pool2d` -> New Version (WIP)
- `max_pool2d_legacy` -> Old version (Stable)